### ISSUE TITLE
Successfully run Redshift example notebook

### ISSUE THREAD

**Issue by SaOgaz** [spacetelescope/dat_pyinthesky#12]
https://github.com/spacetelescope/dat_pyinthesky/tree/master/jwst_nb_viz/example
This could be in either jupyterlab or jupyter notebook

Jira issue: https://jira.stsci.edu/browse/DATJPP-4

- [x] Lim -- Tried and failed. Reported failure at astropy/astroquery#1412
- [x] Busko
- [x] Earl
- [x] Ogaz
- [x] Averbukh
- [x] Geda
- [x] Jones
- [x] Robitaille
- [x] Ferguson

**Comment by ibusko** [comment]
Issues to solve:
- how to install glue-jupyter: it's not conda or pip-installable. Building from sources using the suggested command 'pip install -e . ' requires 'npm', which, even present in the environment, causes a build error. Using 'python setup.py install' causes this error:

Searching for ipymaterialui@ git+https://github.com/maartenbreddels/ipymaterialui#egg=ipymaterialui
Reading https://pypi.org/simple/ipymaterialui/
Couldn't find index page for 'ipymaterialui' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.org/simple/
No local packages or working download links found for ipymaterialui@ git+https://github.com/maartenbreddels/ipymaterialui#egg=ipymaterialui
error: Could not find suitable distribution for Requirement.parse('ipymaterialui@ git+https://github.com/maartenbreddels/ipymaterialui#egg=ipymaterialui')

**Comment by SaOgaz** [comment]
@ibusko I get a similar problem.  Installed npm, cloned glue-jupyter, and still get the error that it needs npm when I try and "pip install -e ."

```
    running jsdeps
    `npm` unavailable.  If you're running this command using sudo, make sure `npm` is available to sudo
    rebuilding js and css failed
    missing files: ['/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/bqplot/static/extension.js', '/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/bqplot/static/index.js']
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/setup.py", line 219, in <module>
        setup(**setup_args)
      File "/Users/ogaz/miniconda3/envs/pyinsky/lib/python3.6/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/Users/ogaz/miniconda3/envs/pyinsky/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/Users/ogaz/miniconda3/envs/pyinsky/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/Users/ogaz/miniconda3/envs/pyinsky/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/setup.py", line 88, in run
        raise e
      File "/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/setup.py", line 81, in run
        self.distribution.run_command('jsdeps')
      File "/Users/ogaz/miniconda3/envs/pyinsky/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/setup.py", line 156, in run
        raise ValueError(msg)
    ValueError: Missing file: /private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/bqplot/static/extension.js
    npm is required to build a development version of widgetsnbextension
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/w_/t1dtpm391bq6vyvhf6_lmn4m0003cg/T/pip-install-x7amirvz/bqplot/
```

**Comment by pllim** [comment]
On Windows, at one point, I had to install nodejs straight into the system path. But OSX, I dunno.

**Comment by SaOgaz** [comment]
@ibusko I'm gonna try following the instructions here https://johnpapa.net/how-to-use-npm-global-without-sudo-on-osx/

**Comment by ibusko** [comment]
@SaOgaz I re-confirmed the same errors under my 10.13.6  desktop. Yesterday I was trying it on a laptop with 10.14.x and got the same errors.

**Comment by ibusko** [comment]
Updating to glue-core-0.14.2 solved the icon problem. It seems to be running correctly now.

**Comment by SaOgaz** [comment]
@ibusko I got a fix for that, put a PR into the notebook for it

**Comment by hcferguson** [comment]
Tried Sara's fix and after restarting and commenting and uncommenting the alternate lines below at random a few times it worked...
```
#specplot.button_action.value = 'brush x'
specplot.widget_button_interact.value = specplot.interact_brush_x
```

**Comment by nmearl** [comment]
This worked for me with no issues, however, basic setup will only allow you to run it in a Jupyter Notebook. If you wish to run it in JupyterLab, you'll have to install the dedicated JupyterLab extension using `jupyter labextension install @jupyter-widgets/jupyterlab-manager`.

**Comment by ibusko** [comment]
This is still not working for me. I get past the previous error (fixed by @SaOgaz) but after that, a call to ipywidgets.VBox() results in the same error as I have seen before: 

AttributeError: module 'bqplot' has no attribute 'ScatterGL'

It could be a version mismatch. @nmearl , could you provide a listing (conda list) of your conda environment? I believe you are on OSX, right?

**Comment by nmearl** [comment]
@ibusko I don't use conda to manage my environment installs, generally. Here is my current pip environment: https://gist.github.com/nmearl/68b894298633071aa16f388f3dfe51a0.

**Comment by ibusko** [comment]
I tried the Quick Install installation page instructions and got the script to run in non-interactive mode. The interactive widget seems to display correctly, but no spectral data is displayed. I can print the spectrum right before the interctive widget is called and it looks correct, but the axis are not populated with anything.

**Comment by pllim** [comment]
```python
spec_file = SDSS.get_spectra(plate=6497, mjd=56329, fiberID=139)[0]
```
gave me
```
HTTPError: HTTP Error 404: Not Found
```

**Comment by brechmos-stsci** [comment]
@astrofrog @eteq 

So, I am trying to do this under linux and everything runs *except*:
![image](https://user-images.githubusercontent.com/18348847/55814020-c9a61600-5abb-11e9-8a04-80918896593d.png)

Then when I go in the console I see:

![image](https://user-images.githubusercontent.com/18348847/55813991-bd21bd80-5abb-11e9-9261-5b3b01c90e99.png)

I installed all the things based on the writeup that @SaOgaz did.  I also tried going in and doing the
`jupyter labextension install bqplot` from `https://github.com/bloomberg/bqplot`. 

Not sure if this is a Linux issue or a Jupyter Lab issue (not sure if others did this one in Jupyter Lab vs Jupyter Notebook).
